index.php

```diff
@@ -65,27 +65,33 @@ function set_default_quantity_for_tickets() {
         );
     }
 
-	/**
-	 * Check if the current post is a valid post type for tickets
-	 *
-	 * @since 1.0.0
-	 *
-	 * @return bool
-	 */
-	function is_valid_post_type() {
-		// bail out if not on a Single Event page.
-		if ( ! is_single() ) {
-			return false;
-		}
-
-		if ( ! class_exists( 'Tribe__Tickets__Main' ) ) {
-			return false;
-		}
-
-		if ( ! in_array( get_post_type(), Tribe__Tickets__Main::instance()->post_types() ) ) {
-			return false;
-		}
-
-		return true;
-	}
+    /**
+     * Validates if the current post type is valid for ticketing.
+     *
+     * This function checks if the current post type is among the post types
+     * managed by the Tribe Tickets plugin.
+     *
+     * @since 1.0.0
+     *
+     * @return bool True if the post type is valid, false otherwise.
+     */
+    function is_valid_post_type() {
+        // Ensure the Tribe__Tickets__Main class exists before proceeding
+        if ( ! class_exists( 'Tribe__Tickets__Main' ) ) {
+            return false;
+        }
+
+        // Get the current post type of the page
+        $post_type = get_post_type();
+
+        // Get the post types from the Tribe__Tickets__Main instance
+        $valid_post_types = Tribe__Tickets__Main::instance()->post_types();
+
+        // Check if the current post type is valid
+        if( ! in_array( $post_type, $valid_post_types ) ) {
+            return false;
+        }
+
+        return true;
```

Comment:
```suggestion
        // Check if the current post type is valid
        return in_array( $post_type, $valid_post_types );
```
Simpler

Reply:
All we've really done here though is:
remove the `is_single()` check

All the other logic is essentially the same. Which begs the question: does this actually solve anything?

Reply:
The `is_single()` check was returning false and not applying the extension change in other post types.